enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,199 @@
+For this cluster administrators want to hide developer sub-catalog (types) 
+without disabling a full console plugin, for example by uninstalling an operator.
+By hiding for example "Service Backed operators", "Helm", etc. 
+they can still install them via the CLI, see their resources in the topology, 
+but don't show them in the add from developer catalog flow.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide Developer catalog for all users.
+For example for customers who has their own way to 
+install workloads, services, etc. without the developer catalog.
+
+#### Story 2
+
+As a cluster admin, I want to hide certain sub-catalogs (types) for all users.
+For example for customers who don't want to use Helm charts, 
+or another sub-catalog without disabling Helm support in the whole developer console.
+
+### Goals
+
+- Allow cluster admins to hide the developer catalog or certain sub-catalogs (types).
+- Provide information to admins about sub-catalogs (types) they can hide from users.
+
+### Non-Goals
+
+- Allow cluster admins to customize categories or hide individual items
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`. 
+We can add 
+`spec.customization.developerCatalog.disabledTypes` which lists types to hide.
+
+### Workflow Description
+
+```text
+├── spec
+│   ├── customization
+|       ├── developerCatalog 
+|           ├── disabledTypes
+|               └──       
+└── ...
+```
+
+Each catalog conforms to the following schema:
+
+// +optional
+- `disabledTypes: string[]` , defines a list of types to hide. 
+Types will vary based on the operator installed.
+
+### API Extensions
+
+Extend the existing `Console` CRD to add `spec.customization.developerCatalog.disabledTypes` which 
+can have list of types disabled. 
+
+So the example yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec: 
+  customization:
+    developerCatalog:
+      disabledTypes: 
+        - Devfile
+        - HelmChart
+  ...
+```
+
+If all the available types are listed in `spec.customization.developerCatalog.disabledTypes` 
+then the complete developer catalog will be disabled. 
+A `ConsoleYAMLSample` snippet will be provided for disabling types in the console YAML editor.
+
+### Risks and Mitigations
+
+**Save disabled**: Saving just the disabled types would 
+show (expected or unexpected) new type if a new operator is installed.
+
+### Drawbacks
+
+N/A
+
+## Design Details
+
+Sub-catalogs (types) could be added dynamically by console plugins, 
+for example when installing an operator.
+
+The type is an identifier specified within the extension list of a dynamic plugin.
+
+We will provide a configuration UI and a yaml snippet in the 
+console to provide a list of these dynamic available sub-catalogs / types.
+
+### Test Plan
+
+Testing will be carried out with the usual Console unit and e2e test suites.
```

Comment:
What specific test cases will you want QE to automate?

Reply:
Added the test cases